File: leetcode/easy/0226InvertBinaryTree/invert_tree.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class IterativeSolution(object):
    def invertTree(self, root):
        if root == None:
            return None
        queue = []
        queue.append(root)

        while (len(queue) != 0):
            cur = queue.pop(0)
            temp = cur.left
            cur.left = cur.right
            cur.right = temp
            if (cur.left != None):
                queue.append(cur.left)
            if (cur.right != None):
                queue.append(cur.right)
        return root

class RecursiveSolution(object):
    def invertTree(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        if root == None:
            return None
        right = self.invertTree(root.right)
        left = self.invertTree(root.left)
        root.left = right
        root.right = left
        return root
        

File: leetcode/easy/0226InvertBinaryTree/invert_tree.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 */

data class TreeNode(var `val`: Int) {
    var left: TreeNode? = null
    var right: TreeNode? = null
}

class IterativeSolution {
    fun invertTree(root: TreeNode?): TreeNode? {
        if (root == null){
            return null
        }
        val queue: Queue<TreeNode> = LinkedList<TreeNode>()
        queue.add(root)

        while(queue.size != 0){
            val current = queue.remove()
            val temp = current.right
            current.right = current.left
            current.left = temp

            if (current.right != null){
                queue.add(current.right)
            }
            if (current.left != null){
                queue.add(current.left)
            }
        }
        return root
    }
}


class RecursiveSolution {
    fun invertTree(root: TreeNode?): TreeNode? {
        if (root == null){
            return null
        }    
        var right = invertTree(root.right)
        var left = invertTree(root.left)
        root.left = right
        root.right = left
        return root
    }
}